### DISCUSSION TITLE
Post-breakpoint split aftermath

### DISCUSSION THREAD

**Discussion by jaril** [RecordReplay/devtools#5190] [General]
Documenting the feedback we've gotten after the most recent breakpoint vs print statement split work that we did.

**Comment by jaril** [comment]
Andarist:

Maybe it's just me and I got used to the previous interface but how breakpoints work now is not intuitive for me. You can click on a line number and get a breakpoint and this will become a regular breakpoint, to make it a logpoint I have to click +, I can remove the breakpoint and the log point stays there - for a very long time I didn't know how to close the logpoint's "window" (the original + becomes -, but you have to hover over that line to see it).
Some interactions are also weirdish because the logpoint is, sort of, a breakpoint and yet:
It's a conceptual clash for me - I can still next/prev in the logpoint's window but i can't do the same using "global" controls
A logpoint without the associated breakpoint also looks a little bit broken to me, although it's not - I probably feel this way cause it's different from other web debuggers that I've been using
And the fact that an enabled breakpoint in the sources panel still doesnt come with a contextual menu with action items like: disable/remove/continue to here is still super confusing cause I try to reach for it regularly. And especially when I couldnt figure out the split between breakpoints and logpoints after coming back to Replay after this redesign
Being able to jump between hits of a breakpoint was one of the most useful things for me when using Replay and to get to those I need to convert a breakpoint to a logpoint, so it's somewaht a chore for me to the point that I never want a "bare" breakpoint. I can live with logs not being enforced as often I was ending up with tooo many logs and it was hard to look through all of that, so perhaps this change was good.
But - I still end up with too many logs as I end up creating logpoints instead of bare breakpoints anyway, just to get access to the "hits timeline"

**Comment by jasonLaster** [comment]
cc @Andarist

**Comment by jonbell-lot23** [comment]
March 15, 2022:

@Andarist I'd be curious to know how things feel after several weeks. Still strange? Seeing any benefits to the new approach? If it were up to you, would you roll this back entirely?

**Reply by Andarist** [reply]
I certainly don't have a full picture of how this actually works, I might miss some implications etc. However, I still find myself tripping over this each time I use Replay. It just doesn't quite click for me and I often reach out for the other thing than I should and find myself confused. I quickly realize now how to fix my mistakes but it's far from intuitive to me. This might very well be just me though. Happy to discuss this further if you have any specific questions or improvement proposals.

**Reply by jonbell-lot23** [reply]
Thanks! That's a great datapoint for us to chew on.
